packages/itwinui-css/src/tile/tile.scss

```diff
@@ -97,6 +97,23 @@
   flex-grow: 2;
   min-width: 0;
 
+  .iui-disabled {
+    filter: grayscale(100%);
+    cursor: not-allowed;
+  }
+
+  .iui-negative {
```

Comment:
The expectation is, that you would need to put tile status class only on root and it would get styling.. With your current code, you need to put on all the sections, which is too much.
Try moving these up. I have a feeling, that setting color of text on root might be enough for it to be inherried in all places.